pkg/operator/ceph/object/spec.go

```diff
@@ -158,11 +158,25 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 		ServiceAccountName: serviceAccountName,
 	}
 
+	opsLogFile := ""
+	if opsLogSidecar := c.store.Spec.Gateway.OpsLogSidecar; opsLogSidecar != nil {
+		// Generate ops log file name based on rgw daemon ID
+		opsLogFile := fmt.Sprintf("ops-log-%s.log", getDaemonName(rgwConfig))
+
+		// Add the side-car container named ops-log
+		podSpec.Containers = append(podSpec.Containers,
+			*controller.RgwOpsLogSidecarContainer(opsLogFile,
+				c.clusterInfo.Namespace, *c.clusterSpec,
+				opsLogSidecar.Resources))
+	}
```

Comment:
There is also an interesting corner case here that I hadn't thought of until now. How does this handle multiple RGWs per node? Each one shares the same ID, so I think both RGWs would log to the same file. Is that a problem? I'm guessing it's fine, but it is worth noting.

If it's a problem, we could update this to have the log file use the K8s Pod ID hash as part of its name, ensuring each pod instance gets a new log file.

Reply:
nope, that's actually nice here, you get all the operations logs from multiple instances of RGW at once place in a file, that's how it works for Ceph as well right now. :)